python/paddle/fluid/executor.py

```diff
@@ -225,7 +226,19 @@ def run(self,
             feed_var_name='feed',
             fetch_var_name='fetch',
             scope=None,
-            return_numpy=True):
+            return_numpy=True,
+            use_program_cache=False):
```

Comment:
I would remove this 'option' and use 'True' by default. Then as a quick follow up, auto-detect the program change (preferred) and auto invalidate cache.

Reply:
currently, the cache cannot check if the program has changed, so user need to set this flag if they are sure the program has not changed, so we have to set this to false by default.